src/App.tsx

```diff
@@ -15,36 +17,106 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  SORT_ALPHABET = 'SORT_ALPHABET',
+  SORT_LENGTH = 'SORT_LENGTH',
+}
+
+type Props = {
+  sortField: SortType;
+  isReversed: boolean;
+};
+
+function getSortGoods(goods: string[], { sortField, isReversed }: Props) {
+  let prepGoods = [...goods];
+
+  if (sortField) {
+    prepGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_ALPHABET:
+          return good1.localeCompare(good2);
+        case SortType.SORT_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    prepGoods = prepGoods.reverse();
+  }
+
+  return prepGoods;
+}
+
+export const App: React.FC<Props> = () => {
+  const [isReversed, setIsReversed] = useState(false);
+  const [sortField, setSortField] = useState<SortType | ''>('');
```

Comment:
The initial state for `sortField` is set to an empty string, which is not compatible with the `SortType` type. Consider initializing it with a valid `SortType` value or adjusting the type definition to include an empty string as a possible value.